src/main/java/spiffe/api/svid/WorkloadAPIClient.java

```diff
@@ -0,0 +1,99 @@
+package spiffe.api.svid;
+
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spiffe.api.svid.retry.RetryHandler;
+import spiffe.api.svid.retry.RetryPolicy;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static spiffe.api.svid.Workload.*;
+
+/**
+ * Provides functionality to interact with a Workload API
+ *
+ */
+public final class WorkloadAPIClient {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(WorkloadAPIClient.class);
+
+    private SpiffeWorkloadStub spiffeWorkloadStub;
+
+    private RetryHandler retryHandler;
+
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress
+     */
+    public WorkloadAPIClient(String spiffeEndpointAddress) {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(spiffeEndpointAddress);
+        retryHandler = new RetryHandler(new RetryPolicy(1, 60, TimeUnit.SECONDS));
+    }
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress
+     */
+    public WorkloadAPIClient(String spiffeEndpointAddress, RetryPolicy retryPolicy) {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(spiffeEndpointAddress);
+        retryHandler = new RetryHandler(retryPolicy);
+    }
+
+    /**
+     * Default constructor
+     *
+     */
+    public WorkloadAPIClient() {
+        this(null);
+
+    }
+
+    /**
+     * Fetch the SVIDs from the Workload API on a asynchronous fashion
+     *
+     */
+    public void fetchX509SVIDs(Consumer<List<X509SVID>> listener) {
```

Comment:
The method should have another name, as it does not fetch SVIDs, but registers a listener. Why not name it `registerListeners`?